backend/src/main.rs

```diff
@@ -55,12 +60,14 @@ fn router(cors: CorsLayer) -> Router {
     frontend.pop();
     frontend.push(FRONTEND_FOLDER);
     tracing::info!("serving frontend at: {frontend:?}");
-    let mut router = Router::new()
+    let router = Router::new()
         // Frontend
         .nest_service("/", ServeDir::new(frontend));
 
     // Add API
-    router = new_api_route!(router, "/root", get(root)).expect("Couldn't register new API route");
+    let router = router
+        .api_route::<_, _, NoApi, ApiDoc>("/root", Method::GET, root)
```

Comment:
This looks ugly. You can add a transparent wrapper (wrapper that can be derefed to original `Router`), that captures `NoApi` and `ApiDoc` as a type parameters and add `api_route` as a function of that wrapper. Wrapper by itself can be created with extension method.
In this case the code can look like:
```Rust
let router = router.with_api_info_wrapper::<NoApi, ApiDoc>() // <- this creates struct ApiInfoWrapper<TApi, TApiDoc>
    .api_route("/root", Method::GET, root)   // <- this adds a new route, performs checks and adds error inside internal vector
    .api_route("/root2", Method::GET, root2) // <- same here
    .unwrap().  // <- returns Result<Router, RouterError> consuming the wrapper
    .expect("Couldn't register new API route"); // <- validation that there are no errors
```

Reply:
Now router supports both new syntax with wrapper and old ugly syntax without one. The reason, why the old one wasn't removed is that it can be used in some situations when you want to just add one API route alongside with regular routes like this: 

```rust 
Router::<(), Body>::new()
    .route("/route", get(route)
    .api_route::<_, _, NoApi, TestDoc>("/api_route", Method::GET, api_route).
    .route("/route2" get(route2))
    .route("/route3" get(route3));
```

Alternative to this is:

```rust
Router::<(), Body>::new()
    .route("/route", get(route).
    .with_context::<NoApi, TestDoc>()
    .api_route("/api_route", Method::GET, api_route)
    .no_context()
    .route("/route2" get(route2))
    .route("/route3" get(route3));
```
The first variant (ultra ugly turbofish one) can be removed if requested

Reply:
First of all, we should not use regular registration, because we always want to validate our routes. Second, the regular routes still can be added after `.expect("Couldn't register new API route")`. So I suggest to remove the old way of registration